src/kbc_modules/cc_kbc/crypto/mod.rs

```diff
@@ -70,59 +69,53 @@ impl TeeKey {
     }
 }
 
-#[derive(Serialize, Deserialize, Debug, Clone)]
-pub struct CryptoAnnotation {
-    algorithm: String,
-
-    // The input to a cryptographic primitive being used to provide the initial state.
-    // Base64 encoded.
-    // If don't need, left blank.
-    #[serde(rename = "initialization-vector")]
-    initialization_vector: String,
-
-    // The symmetric key encrypted by TEE's public key.
-    // This symmetric key is used to encrypt API output by KBS.
-    // Base64 encoded.
-    #[serde(rename = "enc-symkey")]
-    enc_symkey: String,
+#[derive(Serialize, Deserialize)]
+struct ProtectedHeader {
+    // enryption algorithm for encrypted key
+    alg: String,
+    // encryption algorithm for payload
+    enc: String,
 }
 
-impl CryptoAnnotation {
-    // Use the TEE private key to unwrap the encrypted symmetric key,
-    // then use the symmetric key to decrypt cipher text.
-    #[allow(unused_assignments)]
-    pub fn decrypt(&self, tee_key: TeeKey, cipher_text: Vec<u8>) -> Result<Vec<u8>> {
-        let wrapped_symkey: Vec<u8> = base64::decode(self.enc_symkey.clone())?;
-        let symkey: Vec<u8> = tee_key.decrypt(wrapped_symkey)?;
-
-        // Support various algorithm.
-        let plain_text = match self.algorithm.as_str() {
-            AES_GCM_256_ALGORITHM => {
-                let decrypting_key = aes_gcm::Key::<Aes256Gcm>::from_slice(&symkey);
-                let aes_gcm_cipher = Aes256Gcm::new(decrypting_key);
-
-                let iv_decoded = base64::decode(self.initialization_vector.clone())?;
-                let nonce = aes_gcm::Nonce::from_slice(&iv_decoded);
-
-                aes_gcm_cipher
-                    .decrypt(nonce, cipher_text.as_ref())
-                    .map_err(|e| anyhow!("AES_GCM_256_ALGORITHM: {:?}", e))?
-            }
-            _ => {
-                return Err(anyhow!("Unsupported algorithm: {}", self.algorithm.clone()));
-            }
-        };
-
-        Ok(plain_text)
+pub fn decrypt_response(response: &Response, tee_key: TeeKey) -> Result<Vec<u8>> {
```

Comment:
consider making this function a method on Response:

```rust
impl Response {
    fn decrypt(self, tee_key: TeeKey) -> .... {
         ...
    }
}
```

(but maybe that'll make you move it away from the `crypto` mod. not sure)

Reply:
Yeah this was just to keep things together in the crypto file. Might be able to impl from there if I make a bunch of stuff public.

Reply:
Just leave it in `crypto.rs`.